src/logic/community/compliance-validator.ts

```diff
@@ -0,0 +1,98 @@
+import { FeatureFlag } from '../../adapters/feature-flags'
+import { AppComponents } from '../../types'
+import { errorMessageOrDefault } from '../../utils/errors'
+import { CommunityComplianceError } from './errors'
+
+export interface ICommunityComplianceValidatorComponent {
+  validateCommunityContent(request: { name: string; description: string; thumbnailBuffer?: Buffer }): Promise<void>
+}
+
+export function createCommunityComplianceValidatorComponent(
+  components: Pick<AppComponents, 'aiCompliance' | 'featureFlags' | 'logs'>
+): ICommunityComplianceValidatorComponent {
+  const { aiCompliance, featureFlags, logs } = components
+  const logger = logs.getLogger('community-compliance-validator')
+
+  return {
+    async validateCommunityContent(request: {
+      name: string
+      description: string
+      thumbnailBuffer?: Buffer
+    }): Promise<void> {
+      if (!featureFlags.isEnabled(FeatureFlag.COMMUNITIES_AI_COMPLIANCE)) {
+        logger.info('Skipping AI compliance validation for communities because the feature flag is disabled')
+        return
+      }
+
+      const { name, description, thumbnailBuffer } = request
+      const startTime = Date.now()
+
+      try {
+        logger.info('Starting community content compliance validation', {
+          name,
+          descriptionLength: description.length,
+          hasThumbnail: String(!!thumbnailBuffer)
+        })
+
+        const validationResult = await aiCompliance.validateCommunityContent({
+          name,
+          description,
+          thumbnailBuffer
+        })
+
+        const duration = Date.now() - startTime
+
+        if (!validationResult.isCompliant) {
+          logger.warn('Community content compliance validation failed', {
+            name,
+            issues: validationResult.issues.join(', '),
+            warnings: validationResult.warnings.join(', '),
+            confidence: validationResult.confidence,
+            duration
+          })
+
+          throw new CommunityComplianceError(
```

Comment:
Maybe we should distinguish between errors stating that the community did not pass the check and errors stating that the check could not be performed because of any reason